mass_mailing_import_mailing_list/wizard/import_mailing_list.py

```diff
@@ -0,0 +1,360 @@
+import base64
+import csv
+import io
+import logging
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+from odoo.modules.module import get_resource_path
+from xlrd import open_workbook
+
+_logger = logging.getLogger(__name__)
+
+class ImportMailingList(models.TransientModel):
+    """
+    Import knime mailing list.
+    If 'ADKd campaign' is selected we will create
+    a parent mailing list for all the imported mailing lists.
+    We do this in order to be able to unsubscribe to all the
+    ADKd campaign new letters from the imported file. If a
+    customer unsubscribes to ADKd Campaign list then the customer
+    will not be imported to any of the children list next time.
+    """
+    _name = 'import.mailing.list'
+
+    file = fields.Binary(string='File')
+    filename = fields.Char(string='File name')
+    nr_total_rows = fields.Integer(string='Total number of rows')
+    nr_imported_rows = fields.Integer(string='Successful rows')
+    nr_failed_rows = fields.Integer(string='Failed rows')
+    import_failed_mail_ids = fields.One2many(
+                                    comodel_name='import.failed.mail',
+                                    inverse_name='import_id')
+    is_imported = fields.Boolean()
+    import_type = fields.Selection(
+        string="Import type",
+        selection=[("adkd", "ADKd Campaign"),
+                   ("single_use", "Single-use mailing")],
+    )
+    file = fields.Binary(string='File')
+    filename = fields.Char(string='File name')
+    dwnld_csv_filename = fields.Char("CSV File name")
+    dwnld_xls_filename = fields.Char("XLS File name")
+    dwnld_txt_filename = fields.Char("TXT File name")
+    dwnld_csv_file = fields.Binary("CSV File")
+    dwnld_xls_file = fields.Binary("XLS File")
+    dwnld_txt_file = fields.Binary("TXT File")
+
+
+    @staticmethod
+    def check_header(header):
+        """
+        Verify that header has the required columns as well as create
+         a dict with indexes.
+         """
+        # Verify Header, Force it lowercase
+        header = {x.lower(): index for index, x in enumerate(header)}
+        correct_header = [
+            'activemail',
+            'sökande id'
+        ]
+        if not all(item in header for item in correct_header):
+            raise UserError(_('Please correct Header in file!\n'
+                              'Header has to contain:\n') +
+                            '\n'.join(correct_header))
+        return header
+
+    def get_contact(self, sokande_id, partner_obj, contact_obj, email):
```

Comment:
As you are using sudo consider declaring this method as private as odoorpc can call this method otherwise.
odoorcp can call all public methods and thus bypass user rights restrains we want to have in place